pkg/flow/limiter.go

```diff
@@ -0,0 +1,59 @@
+package flow
+
+import (
+	"time"
+
+	"github.com/sirupsen/logrus"
+)
+
+const initialLogPeriod = time.Minute
+const maxLogPeriod = time.Hour
+
+var cllog = logrus.WithField("component", "capacity.Limiter")
+
+// CapacityLimiter forwards the flows between two nodes but checks the status of the destination
+// node's buffered channel. If it is already full, it drops the incoming flow and periodically will
+// log a message about the number of lost flows.
+type CapacityLimiter struct {
+	droppedFlows int
+}
+
+func (c *CapacityLimiter) Limit(in <-chan []*Record, out chan<- []*Record) {
```

Comment:
I'm not sure it's worth it to have a whole stage with in/out buffers for just this simple check, no ? I tried benchmarking against another version with the limiter done at the source and got slightly better time: 15016 ns/op versus 21698 ns/op with a limiter stage (no change in number of allocs)

Reply:
Yeah, I just wanted to separate each functionality as much as possible to avoid overloading too much some pipeline stages. Also because otherwise we should implement this in multiple places: since the deduper is optional, it should Go both in the FlowAccounter and MapTracer.

About the performance improvement, I think in this case is negligible as the pipeline processes in batches and each batch is usually forwarded every `CacheActiveTimeout` period (`5s` by default).

Reply:
fair enough, after our discussion I understand better.
Anyway we can still revisit later if we find there's optimizations to bring here